src/main/java/com/glodblock/github/common/tile/TileFluidPacketDecoder.java

```diff
@@ -164,4 +166,65 @@ public void closeInventory() {
     public boolean isItemValidForSlot(int slot, ItemStack stack) {
         return stack.getItem() instanceof ItemFluidPacket;
     }
+
+    @Override
+    public int fill(ForgeDirection from, FluidStack resource, boolean doFill) {
+        return 0;
+    }
+
+    @Override
+    public FluidStack drain(ForgeDirection from, FluidStack requestFluid, boolean doDrain) {
+        return this.drain(requestFluid, doDrain);
+    }
+
+    private FluidStack drain(FluidStack requestFluid, boolean doDrain) {
+        if (requestFluid == null || requestFluid.amount <= 0) return null;
+        ItemStack fluidPacket = this.inventory.getStackInSlot(0);
+        if (fluidPacket != null) {
+            FluidStack fs = ItemFluidPacket.getFluidStack(fluidPacket);
+            if (fs.isFluidEqual(requestFluid)) {
```

Comment:
all fluidstack from packet should check null.

Reply:
fixed